### ISSUE TITLE
Add an `ariaLabel` property to TimelineItem

### ISSUE THREAD

**Issue by eamodio** [microsoft/vscode#95359]
Refs: #92342, https://github.com/microsoft/vscode/issues/92342#issuecomment-610076915

While implementing #92342 for the Timeline view, I was thinking we should add an optional property to each `TimelineItem` for accessibility called `ariaLabel`. The default of just a concatenation of label and timestamp doesn't feel rich/accurate enough. Whereas a specific `TimelineProvider` can provide a richer, more accurate description for an item. For example, the Git TimelineProvider can provide a more meaningful combination of timestamp, author, commit message (maybe all of it rather than just the first line?), etc, e.g. `3 days ago, <user> committed <commit-message>`

/cc @jrieken @isidorn Thoughts on the idea and also naming?

**Comment by isidorn** [comment]
Yes I think exposing the `ariaLabel` to extensions makes sense. Since we can not always use the `label` as it is not precise enough. `ariaLabel` should be mandatory however if not provided we could fall back to `label`.
Looking back we should have also done it for statusBarItems, since we have an issue there that we support extensions putting icons in the label and icons are obviously pretty useless to screen reader users.

As for aria labels in general my advice is to always present first the most important part of information and then the less important stuff. Reason being: users usually like skipping the rest of whati is pronunced so they first want to hear the important stuff. It should also be concise.
So for me the following works best
`<user> <commit-message>, 3 days ago`

**Comment by isidorn** [comment]
This is actually covered by https://github.com/microsoft/vscode/issues/95360
So let's close as duplicate
